From 88dcc44ea8fbe158d1dee3ea197e47794bf4449d Mon Sep 17 00:00:00 2001
From: Arunk Khandavalli <akhandav@codeaurora.org>
Date: Mon, 12 Jun 2017 14:49:45 +0530
Subject: qcacld-2.0: Specify policy for txpower_scale_decr_db

qcacld-3.0 to qcacld-2.0 propogation

__wlan_hdd_cfg80211_txpower_scale_decr_db() calls nla_parse without
specifying a policy. Specify an nla policy with the expected attribute
information when calling nla_parse in
__wlan_hdd_cfg80211_txpower_scale_decr_db().

Change-Id: I219190726ccb19cb8849177888bca859da1c76db
CRs-Fixed: 2059715
---
 CORE/HDD/src/wlan_hdd_cfg80211.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/CORE/HDD/src/wlan_hdd_cfg80211.c b/CORE/HDD/src/wlan_hdd_cfg80211.c
index 64d7006..f032d04 100644
--- a/CORE/HDD/src/wlan_hdd_cfg80211.c
+++ b/CORE/HDD/src/wlan_hdd_cfg80211.c
@@ -11314,6 +11314,11 @@ static int wlan_hdd_cfg80211_txpower_scale(struct wiphy *wiphy,
 	return ret;
 }
 
+static const struct nla_policy txpower_scale_decr_db_policy
+[QCA_WLAN_VENDOR_ATTR_TXPOWER_SCALE_DECR_DB_MAX + 1] = {
+	[QCA_WLAN_VENDOR_ATTR_TXPOWER_SCALE_DECR_DB] = { .type = NLA_U8 },
+};
+
 /**
  * __wlan_hdd_cfg80211_txpower_scale_decr_db () - txpower scaling
  * @wiphy: Pointer to wireless phy
@@ -11343,7 +11348,7 @@ static int __wlan_hdd_cfg80211_txpower_scale_decr_db(struct wiphy *wiphy,
 	adapter = WLAN_HDD_GET_PRIV_PTR(dev);
 
 	if (nla_parse(tb, QCA_WLAN_VENDOR_ATTR_TXPOWER_SCALE_DECR_DB_MAX,
-		      data, data_len, NULL)) {
+		      data, data_len, txpower_scale_decr_db_policy)) {
 		hddLog(LOGE, "Invalid ATTR");
 		return -EINVAL;
 	}
-- 
cgit v1.1

